app/routes.py

```diff
@@ -1,2 +1,202 @@
+        return {
+            "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+            }, 200
+
+@tasks_bp.route("", methods=["GET"])
+def tasks_index():
+    
+    query_sorted = request.args.get("sort")
+    if query_sorted == "asc":
+        # Found in SQLALchemy documentation. 
+        # The order_by method takes the data in the user table (Task) 
+        # and filters by title in ascending order
+        tasks = Task.query.order_by(Task.title.asc())
+    elif query_sorted == "desc":
+        # Found in SQLALchemy documentation. 
+        # The order_by method takes the data in the user table (Task) 
+        # and filters by title in descending order
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    # This portion is just a GET request
+    if tasks == None:
+        return []
+    else:
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(task.to_json())
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("", methods=["POST"])
+def tasks():
+    try:
+        # This portion is the POST request
+        request_body = request.get_json()
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return {"task": new_task.to_json()}, 201
+    except KeyError:
+        return {
+            "details": "Invalid data"}, 400
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    patch_task = Task.query.get(task_id)
+    date = datetime.utcnow()
+    if patch_task == None:
+        return jsonify(None), 404
+    # Mark Complete on an Incompleted Task
+    patch_task.completed_at = date
+    bot_notification(patch_task)
+    db.session.commit()
+    
+    return {"task":patch_task.to_json()}, 200
+    
+
+def bot_notification(patch_task):
+    PATH = "https://slack.com/api/chat.postMessage"
+    API_TOKEN = os.environ.get("API_KEY")
+
+    query_params = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {patch_task.title}"
+    }
+
+    header = {
+        "Authorization": f"Bearer {API_TOKEN}"
+
+    }
+    result = requests.get(PATH, params=query_params,headers=header)
+
+    return result
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    patch_task = Task.query.get(task_id)
+    if patch_task == None:
+        return jsonify(None), 404
+    patch_task.completed_at = None
+    db.session.commit()
+
+    return {"task":patch_task.to_json()}, 200
+
+
+# ============================Goals=========================================
+
+from app.models.goal import Goal
+from flask import Blueprint, request, jsonify
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
```

Comment:
we could add Goal import to the top. also, we've already imported Blueprint, request, and jsonify, so we don't need to do it again.